common/components/DictionarySettingsTab.tsx

```diff
@@ -130,15 +168,81 @@ const DictionarySettingsTab: React.FC<Props> = ({ settings, onSettingChanged, an
         yomitanSectionRef.current?.scrollIntoView();
     };
 
+    const [buildingAnkiCache, setBuildingAnkiCache] = useState<boolean>(false);
+    const [buildAnkiCacheResult, setBuildAnkiCacheResult] = useState<DictionaryBuildAnkiCacheState | undefined>(
+        undefined
+    );
+
+    const handleBuildAnkiCache = useCallback(async () => {
+        try {
+            setBuildingAnkiCache(true);
+            await dictionaryProvider.buildAnkiCache(activeProfile, settings);
+        } catch (e) {
+            console.error('Failed to send build Anki cache message', e);
+            const message = e instanceof Error ? e.message : String(e);
+            setBuildAnkiCacheResult({ command, msg: message, error: true, modifiedTokens: [] });
+        } finally {
+            setBuildingAnkiCache(false);
+        }
+    }, [dictionaryProvider, settings, activeProfile]);
+
+    useEffect(() => {
+        const removeCB = dictionaryProvider.onBuildAnkiCacheStateChange((state) => {
```

Comment:
Advantage of returning the unsubscription callback is that you can just return it in React `useEffect` functions like this
Not that it matters, just slightly more pleasant to read